### PR TITLE
Task 7 post new comment

### PR DESCRIPTION
**Pull request by benmccarrick**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -217,4 +217,77 @@ describe("GET /api/articles/:article_id/comments", () => {
...
+      .set("Accept", "application/json")
+      .expect(201)
+      .then(({body: {comments}}) => {
+        expect(comments).toMatchObject({
+              comment_id: expect.any(Number),
+              body: 'body of text',
+              author: 'rogersop',
+              article_id: 1,
+              created_at: expect.any(String),
+              votes: expect.any(Number)
```
Comment (ghost) [inline __tests__/app.test.js:238]: Technically if you look at the seed you'll see why votes will be 0

File: __tests__/app.test.js [hunk 2]
```diff
@@ -217,4 +217,77 @@ describe("GET /api/articles/:article_id/comments", () => {
...
+      .send({
+        username: 'invalid_username',
+        body: 'body of text'
+      })
+      .set("Accept", "application/json")
+      .expect(404)
+      .then(({body}) => {
+        expect(body.msg).toBe('Not found');
+      });
+  });
```
Comment (ghost) [inline __tests__/app.test.js:292]: I'd like one more test for sending an empty body, but great effort here.

File: controllers/comments-controller.js [hunk 3]
```diff
@@ -9,4 +10,19 @@ exports.getCommentsByArticleId = (req, res, next) => {
     }).catch((err) => {
         next(err);
     });
+}
+
+exports.addCommentsByArticleId = (req, res, next) => {
+    const articleId = req.params.article_id;
+    const {body, username} = req.body;
+    return getArticleId(articleId)
```
Comment (ghost) [inline controllers/comments-controller.js:18]: This would be a perfect time for a re usable check exists function and a promise.all. Check the advanced error handling.

**Review by ghost — COMMENTED** [review]
Great work on this one. A few areas you could sharpen up.
